ethereum/spec/src/main/java/tech/pegasys/teku/spec/signatures/SigningRootUtil.java

```diff
@@ -112,8 +113,14 @@ public Bytes signingRootForSignVoluntaryExit(
   }
 
   public Bytes signingRootForValidatorRegistration(
-      final ValidatorRegistration validatorRegistration, final UInt64 epoch) {
-    final MiscHelpers miscHelpers = spec.atEpoch(epoch).miscHelpers();
+      final ValidatorRegistration validatorRegistration, final Optional<UInt64> epoch) {
```

Comment:
Personally I'd write this to never take an epoch and always use the genesis miscHelpers.  Like deposits, the validator registration should be valid regardless of forks and that only works if we always use the genesis spec for them.

Reply:
Right! Sorry @usmansaleem for the back and forth :)

Reply:
Sorry for the contradictions.  To be clear there should never be a need to pass `epoch` to `signingRootForValidatorRegistration` as we should always be using the genesis spec when signing to ensure it's consistent across all forks.

Reply:
all good, I've incorporated suggestions. Have a look at your convenience.